### PR TITLE
stop the pills

### PR DESCRIPTION
**Pull request by MrsTonedOne**
|= is slow, as it searches the entire list

list[thing] = thing is much faster.

### CODE REVIEW

File: code/modules/stock_market/stockmarket.dm [hunk 1]
```diff
@@ -1,171 +1,171 @@
-
```
Comment (Iamgoofball) [inline code/modules/stock_market/stockmarket.dm]: uhhh

File: code/modules/stock_market/stockmarket.dm [hunk 2]
```diff
@@ -1,171 +1,171 @@
...
+						output += "<td style='border:none; height: 20px; width: [cwid]px; background:black'>&nbsp;</td>"
+			output += "</tr>"
+			cval -= ost
+		output += "<tr><td style='font-size:10px; height: 20px; width: 100%; background:black; color:green; text-align:center' colspan='[points.len + 1]'>[base_text]</td></tr>"
+	else
+		output += "<tr><td style='width:[width]px; height:[height]px; background: black'></td></tr>"
+		output += "<tr><td style='font-size:10px; background:black; color:green; text-align:center'>[base_text]</td></tr>"
+
+	return "[output]</table>"
+
```
Comment (MrStonedOne) [inline code/modules/stock_market/stockmarket.dm]: Not sure what all this is about. did somebody fuck up the line endings or some shit?

**Comment by Iamgoofball** [comment]
:-1: bluepill or bust

**Comment by Iamgoofball** [comment]
:large_blue_circle: :pill:

File: code/modules/stock_market/stockmarket.dm [hunk 3]
```diff
@@ -1,171 +1,171 @@
...
+/datum/stockMarket
+	var/list/stocks = list()
+	var/list/balances = list()
+	var/list/last_read = list()
+	var/list/stockBrokers = list()
+	var/list/logs = list()
+
+/datum/stockMarket/New()
+		..()
```
Comment (Iamgoofball) [inline code/modules/stock_market/stockmarket.dm]: why is this doubletabbed btw

Comment (MrStonedOne) [inline code/modules/stock_market/stockmarket.dm]: no idea. also, goof

Reminder that when you comment on a line, you should comment below all off the related lines, other wise only one line shows.

your comment should have been 4 or 5 lines down

Comment (xxalpha) [inline code/modules/stock_market/stockmarket.dm]: Will turning iit into an assoc list make other ops slower?

Comment (MrStonedOne) [inline code/modules/stock_market/stockmarket.dm]: its slower then += and makes -= slightly slower. but 5 to 7k things add themselfs to ssobj at world ini, depending on the map, most of those use |=

**Comment by WJohn** [comment]
MORE SPEED.

**Comment by Razharas** [comment]
When will lummox let people use simple lists for ordering and binary search already

**Comment by MrStonedOne** [comment]
My very unscientific testing is producing rather odd results.

It's always faster, but it's between 1 to 15 seconds faster.

Must be related to cpu cache luck.

**Comment by X-TheDark** [comment]
Is there a difference between using `list -= foo` and `list[foo] = null` ?

**Comment by optimumtact** [comment]
Are you wanting to test this more or are you still happy to merge it for the variable buff of speed anyway?

**Comment by RemieRichards** [comment]
@X-TheDark yes, ```list[foo] = null``` still contains ```foo``` as a key, meaning any for loops (which is the majority of work happening here) over the list will still find it, meaning you've effectively removed nothing.

**Comment by MrStonedOne** [comment]
@optimumtact actually i was thinking i might make it use a var on all atoms or datums, check that, and add/remove using a macro that checks the var. would be even faster and not require list searching or associated list hacks

File: code/game/gamemodes/clock_cult/clock_machines.dm [hunk 4]
```diff
@@ -17,8 +17,8 @@
 		user << "<span class='[powered ? "brass":"alloy"]'>It has access to [powered == INFINITY ? "INFINITY":"[powered]"]W of power.</span>"
 
 /obj/structure/clockwork/powered/Destroy()
-	SSfastprocess.processing -= src
-	SSobj.processing -= src
+	STOP_PROCESSING(SSfastprocess, src)
+	STOP_PROCESSING(SSobj, src)
```
Comment (MrStonedOne) [inline code/game/gamemodes/clock_cult/clock_machines.dm]: todo: fix this line.

File: code/game/gamemodes/clock_cult/clock_machines.dm [hunk 5]
```diff
@@ -34,13 +34,13 @@
...
+			START_PROCESSING(SSfastprocess, src)
 		else
-			SSobj.processing |= src
+			START_PROCESSING(SSobj, src)
 	else
 		icon_state = inactive_icon
-		SSfastprocess.processing -= src
-		SSobj.processing -= src
+		STOP_PROCESSING(SSfastprocess, src)
+		STOP_PROCESSING(SSobj, src)
```
Comment (MrStonedOne) [inline code/game/gamemodes/clock_cult/clock_machines.dm]: todo: fix this line

File: code/game/gamemodes/clock_cult/clock_machines.dm (hunk 4 above)
Comment (RemieRichards) [inline code/game/gamemodes/clock_cult/clock_machines.dm]: whats wrong with it?

**Pull request merged**